library/ecp_internal.h

```diff
@@ -28,6 +28,18 @@
 #include "mbedtls/bignum.h"
 #include "bignum_mod.h"
 
+/** Convert an MPI to its canonical representative.
+ *
+ * \note Currently handles the case when `N->int_rep` is
+ * MBEDTLS_MPI_MOD_REP_MONTGOMERY.
+ *
+ * \param[in,out] X     The address of the MPI to be converted. Must have the
+ *                      same number of limbs as \p N.
```

Comment:
We have stronger requirements than just having the same number of limbs, we only get canonical representative if 0<=X<2N

Reply:
Should that stronger requirement be tested for? It feels like it should in the `.funtion` code. 

The `.is_valid()` in Python is already checking for X < 2N.

Reply:
In general it is a good idea, it makes testing more robust. However in this case I think it is a bit awkward to test it. Maybe `core_sub()` twice and make sure that at least one of them overflows?